airflow/providers/databricks/sensors/databricks_sql.py

```diff
@@ -0,0 +1,136 @@
+    """
+
+    template_fields: Sequence[str] = (
+        "databricks_conn_id",
+        "sql",
+        "catalog",
+        "schema",
+        "http_headers",
+    )
+
+    template_ext: Sequence[str] = (".sql",)
+    template_fields_renderers = {"sql": "sql"}
+
+    def __init__(
+        self,
+        *,
+        databricks_conn_id: str = DatabricksSqlHook.default_conn_name,
+        http_path: str | None = None,
+        sql_endpoint_name: str | None = None,
+        session_configuration=None,
+        http_headers: list[tuple[str, str]] | None = None,
+        catalog: str = "hive_metastore",
+        schema: str = "default",
+        sql: str | None = None,
+        handler: Callable[[Any], Any] = fetch_all_handler,
+        client_parameters: dict[str, Any] | None = None,
+        **kwargs,
+    ) -> None:
+        """_summary_
+
+        :param databricks_conn_id: _description_, defaults to DatabricksSqlHook.default_conn_name
+        :param http_path: _description_, defaults to None
+        :param sql_endpoint_name: _description_, defaults to None
+        :param session_configuration: _description_, defaults to None
+        :param http_headers: _description_, defaults to None
+        :param catalog: _description_, defaults to "hive_metastore"
+        :param schema: _description_, defaults to "default"
+        :param sql: _description_, defaults to None
+        :param handler: _description_, defaults to fetch_all_handler
+        :param client_parameters: _description_, defaults to None
+        """
+        super().__init__(**kwargs)
+        self.databricks_conn_id = databricks_conn_id
+        self._http_path = http_path
+        self._sql_endpoint_name = sql_endpoint_name
+        self.session_config = session_configuration
+        self.http_headers = http_headers
+        self.catalog = catalog
+        self.schema = schema
+        self.sql = sql
+        self.caller = "DatabricksSqlSensor"
+        self.client_parameters = client_parameters or {}
+        self.hook_params = kwargs.pop("hook_params", {})
+        self.handler = handler
+
+    def _get_hook(self) -> DatabricksSqlHook:
+        return DatabricksSqlHook(
+            self.databricks_conn_id,
+            self._http_path,
+            self._sql_endpoint_name,
+            self.session_config,
+            self.http_headers,
+            self.catalog,
+            self.schema,
+            self.caller,
+            **self.client_parameters,
+            **self.hook_params,
+        )
+
+    def _sql_sensor(self, sql):
+        hook = self._get_hook()
+        sql_result = hook.run(
+            sql,
+            handler=self.handler if self.do_xcom_push else None,
+        )
+        return sql_result
+
+    def _get_results(self) -> bool:
+        result = self._sql_sensor(self.sql)
+        self.log.debug("SQL result: %s", result)
+        if len(result) < 1:
+            raise AirflowException("No results for SQL sensor.")
```

Comment:
Sensor could be called multiple times while waiting for results. To have this it's imho better to do `return False` instead of throwing an exception

Reply:
I tried to solve this multiple times before submitting the PR to have it return `False`, but it goes into an infinite loop when set to False (saw this in a few other sensors' code as well). So, I added the exception to be raised, making that task instance to fail. 
I couldn't make it work any other way.

Reply:
it's regulated by timeout on the sensor.

Reply:
That sounds good, I will change it for all the sensors, assuming the DAG will have a timeout for the sensor object.

Reply:
Just pushed the change.